plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +720,69 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+unsigned countWords(UnicodeString const & source, unsigned delimLen, UChar const * delim, bool allowBlankItems)
+{
+    int32_t wordCount = 0;
+    UnicodeString const delimiter(delim, delimLen);
+    if (source.isEmpty() || (delimiter.isEmpty() && delimLen != 0))
+    {
+        return wordCount;
+    }
+    int32_t startWord = NULL;
+    int32_t idx = 0;
+    int32_t sourceLength = source.countChar32();
+    int32_t delimiterLength = delimiter.countChar32();
+    if (sourceLength < delimiterLength)
+    {
+        wordCount = 1;
+        return wordCount;
+    }
+    int32_t max = source.length() - (delimiterLength - 1);
```

Comment:
Despite the "iterate by code point" functions that are in use to account for surrogate pairs, they still measure the string in terms of code units. So if you have a string that is just one surrogate pair character and have the iterator move 1 code point forwards from the start, the iterator's value would equal 2 (code units) instead of 1 (code point).
Actually, the delimiterLength should be changed to delimiter.length(), despite it not actually causing a problem.